lib/media/track/remotevideotrack.js

```diff
@@ -67,33 +81,52 @@ class RemoteVideoTrack extends RemoteMediaVideoTrack {
     });
   }
 
+  /**
+   * @private
+   */
+  _start(dummyEl) {
+    const result = super._start.call(this, dummyEl);
+    updateRenderHints(this);
+    return result;
+  }
+
+
   attach(el) {
     const result = super.attach(el);
     if (this._intersectionObserver !== null) {
+      this._invisibleElements.add(result);
       this._intersectionObserver.observe(result);
+      // NOTE(mpatwardhan): we do not call updateRenderHints here,
+      // because the dimensions are not known for freshly created
+      // elements. we will get non-zero dimensions in _intersectionObserver
+      // callback and then will call updateRenderHints.
     }
+
     if (this._enableDocumentVisibilityTurnOff) {
-      this._documentVisibilityTurnOffCleanup = this._documentVisibilityTurnOffCleanup
-        || setupDocumentVisibilityTurnOff(this);
+      this._documentVisibilityTurnOffCleanup = this._documentVisibilityTurnOffCleanup || setupDocumentVisibilityTurnOff(this);
     }
+
     return result;
   }
 
   detach(el) {
     const result = super.detach(el);
-    if (this._intersectionObserver !== null) {
-      if (Array.isArray(result)) {
-        result.forEach(element => this._intersectionObserver.unobserve(element));
-      } else {
-        this._intersectionObserver.unobserve(result);
+    const elements = Array.isArray(result) ? result : [result];
+    elements.forEach(element => {
+      if (this._intersectionObserver !== null) {
+        this._intersectionObserver.unobserve(element);
       }
```

Comment:
nit: No strong opinion here but this is probably safer no?
```js
      if (this._intersectionObserver) {
        this._intersectionObserver.unobserve(element);
      }
```
Or a one liner. Not sure if this passes linter though.
```js
    this._intersectionObserver && this._intersectionObserver.unobserve(element);
```

Reply:
sure, I will update this.